platform/SingletonPtr.h

```diff
@@ -43,6 +43,10 @@ extern osMutexId_t singleton_mutex_id;
 inline static void singleton_lock(void)
 {
 #ifdef MBED_CONF_RTOS_PRESENT
+    if (!singleton_mutex_id) {
+        // RTOS has not booted yet so no mutex is needed
+        return;
```

Comment:
Should we throw an `MBED_ERROR` here?

Reply:
nope, this case is expected to be hit early in the boot sequence when when the RTOS has not yet started.

Reply:
It does seem to be the case for STM boards...